rust/protocol/src/protocol.rs

```diff
@@ -358,10 +401,10 @@ impl TryFrom<&[u8]> for PreKeySignalMessage {
         let base_key = PublicKey::deserialize(base_key.as_ref())?;
 
         Ok(PreKeySignalMessage {
-            message_version,
-            registration_id: proto_structure.registration_id.unwrap_or(0),
-            pre_key_id: proto_structure.pre_key_id,
-            signed_pre_key_id,
+            message_version: message_version.try_into()?,
```

Comment:
Hmm, we've already checked the version at this point, so this should probably use `expect()` instead of propagating the error.

Reply:
You're right -- great catch. This can just use `.into()` instead.